data_pipeline_api/file_api.py

```diff
@@ -202,7 +202,10 @@ def __init__(self, config_filename: Optional[Union[Path, str]] = None):
         self._run_metadata["run_id"] = self._run_id
         self._run_metadata["data_directory"] = str(self._unnormalised_data_directory)
         self._run_metadata["open_timestamp"] = self._open_timestamp
-        self._run_metadata.update(self._config.get("run_metadata", {}))
+        run_metadata = self._config.get("run_metadata", {})
+        if any(key in run_metadata for key in FileAPI.RESERVED_RUN_METADATA_KEYS):
+            raise ValueError(f"reserved key {key} is set in run_metadata")
```

Comment:
I am now failing if anyone puts a reserved key into the config run_metadata.

Reply:
I should probably do this too

Reply:
with regards to throwing an exception

Reply:
yeah non-critical, as long as we have the same implementation of precedence for where the output run_metadata gets generated, but I do think it would be useful.

Reply:
Ok sure, will address separately then